src/main/scala/rps/GameController.scala

```diff
@@ -17,11 +16,17 @@ trait GameController {
 }
 
 class GameControllerImpl(gameService: GameService)(implicit ec: ExecutionContext) extends GameController {
-  override def result(): Future[Either[Throwable, PlayResponse]] = {
-    gameService.getResult.map(p => p.flatMap(_.toRight(new IllegalStateException).map {
-      play => PlayResponse(play.userMove, play.computerMove, play.result)
-    }))
-  }
+  val runtime = Runtime.default
```

Comment:
I would get the runtime as an implicit parameter. Also I don't think you need the implicit `ExecutionContext` anymore